src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  die(dead) {
+    if (dead.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
Using filter inside the 'die' method is not efficient because it will recreate the alive array every time an animal's health drops below 0. Instead, you should consider using splice to remove the dead animal from the array, which would be more performant. However, since the checklist specifically mentions avoiding nesting loops, ensure that you implement this in a way that does not introduce nested loops.